user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -155,6 +155,7 @@ public static UserOperatorConfig buildFromMap(Map<String, String> map) {
 
         Map<String, Object> generatedMap = ConfigParameter.define(envMap, CONFIG_VALUES);
 
+        // This is to make sure that if CA namespace is empty or null, then we can use the namespace value
         if (generatedMap.get(CA_NAMESPACE.key()) == null || String.valueOf(generatedMap.get(CA_NAMESPACE.key())).isEmpty()) {
             generatedMap.put(CA_NAMESPACE.key(), generatedMap.get(NAMESPACE.key()));
         }
```

Comment:
Shouldn't this be handled in the application code rather then here?

Reply:
I tried to place a good place for it but it seems to fit here. Since the code we use for the generation of the configuration map, its generic so it will be used for topic and cluster operator. So putting it there wouldn't be much useful.

Reply:
The idea was more to handle it where the CA_NAMESPACE is actually used as that place knows best how to handle its values. Another option would be for example to have some method `getCaNamespaceOrNamespace` which will wrap this logic. This will allow you to have it centrally, but not modify the configuration. I think the way you do it right now is questionable because it distorts what is the actual configuration set by the user and how we use the configuration. I think having a clear separation of that is something that we should try to keep.

Reply:
Yeah that sounds good. I will have a look at where it is used

Reply:
FYI: One of the places uses it incorrectly. I already opened https://github.com/strimzi/strimzi-kafka-operator/pull/8235 to fix it there. so you should just ignore that one.